log-appender/src/main/java/io/phdata/pulse/log/HttpAppender.java

```diff
@@ -96,28 +109,44 @@ private void flush() throws Exception {
     }
   }
 
+  /**
+   * Flush only if conditions are met
+   *
+   * @throws Exception
+   */
+  void tryFlush() throws Exception {
```

Comment:
I don't see this being used

Reply:
If this method gets removed, you can also delete 'shouldFlush()' as it is only used here.